### ISSUE TITLE
Looking for one maintainer

### ISSUE THREAD

**Issue by oliviertassinari** [oliviertassinari/react-swipeable-views#558]
I don't have time to dedicate to this project anymore. If somebody would like to keep up the maintenance, I would be happy to consider giving him the necessary right to do so. Thanks

**Comment by jeanpoelie** [comment]
Besides checking pull-requests and reacting to issues what else do you expect from the maintainer? I'd love to help but I do not have in-depth knowledge of the library yet, so root-issues would be a challenge to fix.

**Comment by oliviertassinari** [comment]
I think that it would be great to have somebody that can provide peer reviews to the open pull requests.

**Comment by jeanpoelie** [comment]
Most PR's I see right now seem to be pretty straight forward, some might be somewhat challenging for a junior react dev like myself.

**Comment by oliviertassinari** [comment]
Basically, I wish that we can "unlock" the project from its position of immobility. I think that having somebody that can provide peer reviews and be accountable for the changes would be good enough. From experience, I know that the project can't really expect more and that's fine.

**Comment by jeanpoelie** [comment]
I'd love to take up this task, mostly for 2 PR's that are a must upgrade, resulting in animatieHeight not working and warnings being generated

**Comment by oliviertassinari** [comment]
@jeanpoelie Awesome, I have given you write access to the repository :)

**Comment by vanvuongngo** [comment]
There are two PR to make this library react 16.9 and 17.x ready but I would like to fix it with functional components and hooks instead of adding **UNSAFE_** methods.

**Comment by oliviertassinari** [comment]
@vanvuongngo This would likely require to migrate the tests first.

**Comment by baoduy** [comment]
Hi @oliviertassinari 
I'm happy to be a maintainer for this project. Please added me in. This project also has an issue sine the main file of package.json is pointing to the wrong js file.

**Comment by oliviertassinari** [comment]
@baoduy Thanks for the interest, could you see with @jeanpoelie?

**Comment by jeanpoelie** [comment]
would be awesome to get some extra help, i havent been able to get to the main.js issue

**Comment by caleb-harrelson** [comment]
TL;DR -- I can haz collaborator access to the repo and npm, @oliviertassinari / @jeanpoelie?

I use this package in a limited scope and not in react-native, so my knowledge of it is rather shallow, but in its present state it will block us from updating to React v17. 

I'm not keen on being the sole maintainer, but I can commit to the following in the near term.

# Merging current PRs:
- Merging PR #597 (and closing duplicates #591, #588) as this is the barrier to upgrading to React 17.
- Creating a new PR that implements the new lifecycle methods instead of relying on the deprecated `UNSAFE_componentWillReceiveProps`.
- Merging minor PRs #592, #610, and #612. 
- Analyze #602 to determine if this is the proper way of fixing the animation not working on the first change, merging if so or finding an appropriate fix otherwise.

# Cleaning up other PRs:
- Looking into #585 to see if it's something I can be confident in fixing without breaking anything else, coding/merging/rejecting the PR based on that research.
- Watching #609 for maturity and consider merging.
- Closing #598 due to feature freeze.
- Seeking repro steps on #589 and #614.

# Tidying up:
- Removing ties in the build scripts to Material-UI's documentation in firebase (so that you don't have to give permissions to an unrelated project's firebase account just to facilitate automated documentation on a project you no longer wish to maintain).

Lastly, I saw somewhere in here that @jeanpoelie was having problems getting things built/published. To make sure I didn't run into the same problems I applied #597 manually and built/deployed it to a new package successfully. However, I'd prefer to shut that package down and apply all the above changes here so everyone can benefit.

**Comment by oliviertassinari** [comment]
@caleb-harrelson sounds great :)

**Comment by mixtah** [comment]
Hi @oliviertassinari  I'm using this package for a project (will be running in dev for the next 1-1.5 years) and happy to review pull requests and fix the occasional bug. I can't guarantee a lot of long term support but I'm sure I can fill in the gaps left by others.

**Comment by caleb-harrelson** [comment]
> @caleb-harrelson sounds great :)

Great! What do you need from me to get me added to the project, @oliviertassinari?

**Comment by kodai3** [comment]
@caleb-harrelson 
>Watching #609 for maturity and consider merging.

I wrote #609. I left it immature because I found it takes match more work than I first though and this repo is not so active.
If you work on PRs, I will work on that more actively but I would love to have help or suggestions :)

**Comment by oliviertassinari** [comment]
> Great! What do you need from me to get me added to the project, @oliviertassinari?

Nothing more :), let me invite you as a collaboration

**Comment by nagad814** [comment]
@caleb-harrelson when will the next release happen. I can help with chores if any. Please tag me.

**Comment by caleb-harrelson** [comment]
> @caleb-harrelson when will the next release happen. I can help with chores if any. Please tag me.

I have a looming deadline on Monday, so I plan to tackle it early next week.

**Comment by caleb-harrelson** [comment]
@oliviertassinari, a couple problems in my attempts to maintain the library:

1. My PR (#616) to fix the js warnings is hung on a failed status from codecov.io, despite the [status being good](https://codecov.io/gh/oliviertassinari/react-swipeable-views/commit/9d3667bf43c120617851f98fd314cbeb78630e0d). I can't access the project settings to change anything, so I can't remove the build requirement or any other workarounds (like pushing the PR commit directly to master) to merge the PR.
2. Once I do get it merged, I will not be able to publish the package to NPM without being added to the collaborators, which are currently just you and @jeanpoelie.

Could you please assign the permissions necessary for me to get through these roadblocks?

**Comment by oliviertassinari** [comment]
1. fixed
2. slug?

**Comment by caleb-harrelson** [comment]
> 2. slug?

I presume you're asking for my user on npm? It's https://www.npmjs.com/~caleb-harrelson

**Comment by caleb-harrelson** [comment]
@oliviertassinari we're getting close, I can feel it :p

It looks like I have access to publish the main package now, but not its dependencies or the native package:
- [react-swipeable-views-core](https://www.npmjs.com/package/react-swipeable-views-core)
- [react-swipeable-views-utils](https://www.npmjs.com/package/react-swipeable-views-utils)
- [react-swipeable-views-native](https://www.npmjs.com/package/react-swipeable-views-native)

**Comment by caleb-harrelson** [comment]
@oliviertassinari I still don't have the permissions necessary on npmjs to execute a build successfully. Master is in a state that fixes the immediate issue if you want to issue the build yourself instead of granting the necessary permissions.

**Comment by oliviertassinari** [comment]
@caleb-harrelson I have invited you to the 3 repositories, it should work now.

**Comment by oliviertassinari** [comment]
@caleb-harrelson Any luck with the release of the package?

**Comment by caleb-harrelson** [comment]
@oliviertassinari Thanks for the reminder. I've published it as an alpha for now until I have a little more time to test it.

**Comment by atnpcg** [comment]
@caleb-harrelson  what is left to test? I can help if anything is pending, I want 0.14.0 out of alpha :)

**Comment by caleb-harrelson** [comment]
@atnpcg, that's a great question. My use of the package is very limited, so for the foreseeable future I'm not really able to test beyond what I use them for (i.e. the React component behind the tabbed interface at the bottom of [this page](https://demos.creative-tim.com/material-dashboard-pro-react/#/admin/panels)). I thought it was safer to put a fix out as an alpha that people know might have issues than to issue a new version that isn't adequately tested.

**Comment by vicasas** [comment]
Hello everyone! could I join to help keep this project going? I use it daily in my projects I find it incredible and I would be sorry if it did not continue to advance @oliviertassinari @jeanpoelie

**Comment by oliviertassinari** [comment]
@vicasas I appreciate the proposal. If I see you perform well on Material-UI, with pleasure. I want to make sure whoever is publishing new releases has enough clarity to not introduce regressions at the same time.

**Comment by vicasas** [comment]
@oliviertassinari I understand, no problem :grinning:

**Comment by mayank-budhiraja** [comment]
Hello @oliviertassinari - I am happy to contribute to the repo

**Comment by oliviertassinari** [comment]
@mayank-budhiraja If you could have a look at the open PRs, and continue them to be in a mergeable state, it would be great

**Comment by RedRubyX** [comment]
Hey guys, I allowed myself to update the dependencies and some other details. Please have a look and use the changes in the original, if you like to. In parallel I'm going to publish a package so that my colleagues and I can work with a (hopefully) stable version, since my changes have to be fully tested. 

https://github.com/RedRubyX/react-swipeable-views

**Comment by korbav** [comment]
> Hey guys, I allowed myself to update the dependencies and some other details. Please have a look and use the changes in the original, if you like to. In parallel I'm going to publish a package so that my colleagues and I can work with a (hopefully) stable version, since my changes have to be fully tested.
> 
> https://github.com/RedRubyX/react-swipeable-views

Pretty good, you might want to include this highly requested PR as well : https://github.com/oliviertassinari/react-swipeable-views/pull/654

**Comment by RedRubyX** [comment]
> > Hey guys, I allowed myself to update the dependencies and some other details. Please have a look and use the changes in the original, if you like to. In parallel I'm going to publish a package so that my colleagues and I can work with a (hopefully) stable version, since my changes have to be fully tested.
> > https://github.com/RedRubyX/react-swipeable-views
> 
> Pretty good, you might want to include this highly requested PR as well : #654

@korbav I think this this should be integrated already, since I forked from the developer who made a package from this PR.
